### PR TITLE
[metrics] File type detection for binary files and custom patterns

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR improves file type detection by adding support for identifying common binary executables and compressed files. 

Additionally, it allows to define custom file type patterns for both code and binary files.

### CODE REVIEW

**Review by sduenas — CHANGES_REQUESTED** [review]
I need some examples of possible patterns in the README so users know how to run it.

**Comment by jjmerchante** [comment]
I included one example of each of the possible patterns

File: tests/unit/test_metrics.py [hunk 1]
```diff
@@ -122,8 +122,21 @@ def test_file_type_metrics(self):
...
+        """Test that file type metrics are calculated correctly with new regex"""
+
+        analyzer = GitEventsAnalyzer(code_file_pattern=r"\.py$", binary_file_pattern=r"\.md$")
+
+        analyzer.process_events(self.events)
+
+        file_metrics = analyzer.get_file_type_metrics()
+        self.assertEqual(file_metrics.get("code", 0), 53)
+        self.assertEqual(file_metrics.get("binary", 0), 4)
+        self.assertEqual(file_metrics.get("other", 0), 22)
```
Comment (sduenas) [inline tests/unit/test_metrics.py:138]: We need a test that checks all of them are zero.

File: trustable_cli/metrics.py [hunk 2]
```diff
@@ -190,8 +197,10 @@ def _update_file_metrics(self, event):
             if not file["file"]:
                 continue
             # File type metrics
-            if re.search(FILE_TYPE_CODE, file["file"]):
+            if self.re_code_pattern.search(file["file"]):
```
Comment (sduenas) [inline trustable_cli/metrics.py:200]: When a type of file is not found, the metric is not included. Code, binary and other should always been returned even when files of that type couldn't be found.

**Review by sduenas — CHANGES_REQUESTED** [review]

**Review by sduenas — APPROVED** [review]
LGTM
